feat: #4 - 설정 화면 UI 구성

Gwangyong opened the pull request:
## 🔗 관련 이슈

- #4 

## 🔧 PR 요약
- UITableView + DataSource로 화면을 구성했습니다.
- 커스텀 셀을 통해 "알림 설정", "테마 설정", "문의/건의하기", "리뷰 남기기"를 추가하고, footer에는 앱 버전("ver.1.0.0")을 표시했습니다.
- Accessory로 toggleButton, label, disclosure Indicator를 구성했습니다.

## ✅ 작업 내용 체크리스트

- [x] base 브랜치를 develop으로 설정했나요?
- [x] develop 브랜치를 Pull 받았나요?
- [x] PR의 라벨을 설정했나요?
- [x] assignee를 설정했나요?
- [x] reviewers를 설정했나요?
- [x] 변경 사항에 대한 테스트를 진행했나요?

## 📸 스크린샷 (선택)
![Image](https://github.com/user-attachments/assets/d472e2eb-edc7-4543-afe3-0bc5d2cd17dc)

## 📎 기타 참고사항
- 이후 마이페이지에서 구현할 기능들
  - "알림 설정": 토글(true/false)로 알림 설정 기능 구현 예정
  - "테마 설정": 모달을 통해 "시스템 설정", "다크 모드", "라이트 모드" 전환 기능 구현 예정
  - "문의/건의하기": 웹뷰 이동 기능 구현 예정
  - "리뷰 남기기": 앱스토어 이동 기능 구현 예정

Gwangyong committed:
feat: #4 - 기본적은 UITableView 구성

- UITableView + DataSource로 구현
- separatorInset = .zero로 디자인과 동일한 구분선 구현
- 임시 데이터로 테이블 뷰 구성 확인

Gwangyong committed:
feat: #4 - 커스텀 섹션 헤더 추가

- SettingHeaderView 생성
- 커스템 헤더 추가하여 UI 확인
- #Preview 코드 추가하여 프리뷰 확인 (추후 제거 예정)

Gwangyong committed:
Merge branch 'develop' into feat/4-settings-ui

Gwangyong committed:
feat: #4 - 테이블 헤더 방식 변경 및 스크롤, 레이아웃 수정

- 섹션 헤더 대신 tableHeaderView사용
- 인디게이터 비활성화, 세로 스크롤만 허용
- 가로는 tableView.bounds.width로 고정
- 세로는 UIVIew의 최소 크기로 높이를 오토레이아웃에 맞도록 계산

Gwangyong committed:
feat: #4 - 커스텀 푸터 뷰 추가 및 헤더/푸터 높이 계산 로직 통합

- 확장을 생각해서 SettingFooterView로 파일 생성
- 푸터도 헤더와 동일한 오토레이아웃 기반 높이 계산 적용
- 중복된 헤더/푸터 코드를 updateTableHeaderFooterHeight 메서드로 통합

Gwangyong committed:
chore: #4 - View 폴더링 및 Cell 초기 생성

- View 폴더를 생성하여, View 파일들 정리
- SettingViewCell 파일을 만들어서 초기 세팅

Gwangyong committed:
feat: #4 - 커스텀 셀 UI 및 VC에서 configure

- VC에서 SettingItem enum으로 관리
- configure를 통해 cell에 입력.
- 이미 정의되어 있기에 atributedText로 텍스트, 스타일, 컬러 주입
- 피그마 UI와 동일하게 제약조건 부여

Gwangyong committed:
feat: #4 - 설정 화면에 셀 액세서리 추가

- SettingItem에 accessory 프로퍼티 추가
- UISwitch 토글, Label, disclosure 지원
- configure시에 액세서리 초기화 후 타입에 맞도록 설정
- 토글 스위치는 onTintColor에 지정된 색 설정
- value 라벨은 sizeToFit과 스타일 적용

Gwangyong committed:
refactor: #4 - 설정 화면 전용 모델 분리

- Theme, SettingItem, Accessory를 별도의 파일로 분리

Gwangyong committed:
refactor: #4 - 데이터 분리하여 ViewModel로 이동

- SettingItem 배열을 VC에서 VM으로 이동
- MVVM 구조에 맞게 책임 분리

Gwangyong committed:
feat: #4 - 셀 탭 하이라이팅 .none

rkgus24 commented on PodoIt/Scenes/Setting/View/SettingViewCell.swift:
```diff
@@ -0,0 +1,65 @@
+    
+    switch item.accessory {
+    case .toggle(let isOn):
+      let toggleSwitch = UISwitch()
```
여기서 UISwitch를 프로퍼티로 가지고 상태만 갱신하면 재사용 했을 때 객체 생성이나 이벤트 중복되는 거 막을 수 있을 거 같아요 !

Gwangyong replied:
그러게요 재사용 될때를 생각 못했네요. 바로 반영했습니다! 좋은 피드백 감사합니다 👍

rkgus24 commented on PodoIt/Scenes/Setting/View/SettingViewCell.swift:
```diff
@@ -0,0 +1,65 @@
+  private func configureLayout() {
+    itemLabel.snp.makeConstraints {
+      $0.top.bottom.equalToSuperview().inset(16)
+      $0.leading.trailing.equalToSuperview().inset(20)
```
지금은 UI만 그려서 고려 안 하신 거 같긴 한데 나중에 automaticDimension 한 번 알아보셔도 좋을 거 같네요  👍🏻

Gwangyong replied:
다른 앱들도 살펴보니 보조텍스트 넣어서 셀이 2줄 이상이 되는 곳들이 있어서 말씀하신 `automaticDimension`이 있으면 좋을 것 같네요. 추가했습니다 감사합니다!

rkgus24 commented on PodoIt/Scenes/Setting/View/SettingViewController.swift:
```diff
@@ -0,0 +1,82 @@
+    $0.register(SettingViewCell.self, forCellReuseIdentifier: SettingViewCell.id)
+    $0.separatorInset = .zero // 구분선의 시스템 마진을 zero로
+    $0.alwaysBounceHorizontal = false
+    $0.showsHorizontalScrollIndicator = false
```
UITableView에서는 기본적으로 세로 스크롤만 지원해서 아마 두 줄 지워도 괜찮을 거 같아요

Gwangyong replied:
말씀해주신대로, `alwaysBounceHorizontal`는 필요 없을 것 같아서 삭제했고, `showsHorizontalScrollIndicator = false`는 실제 앱에서는 어차피 안보여서 문제 없지만, Xcode의 View Hierarchy Debugger에 인디게이터가 보이길레 `false`로 처리했습니다.

Gwangyong committed:
feat: #4 - 코드리뷰 피드백 반영. 테이블뷰 속성 정리

- rowHeight를 automaticDimension 설정
- estimatedRowHeight로 추정값 56으로 설정
- 가로 스크롤 인디게이터 숨김 처리 유지
- 불필요한 isDirectionalLockEnabled속성 제거

Gwangyong committed:
refactor: #4 - 피드백 반영. UISwitch 프로퍼티로 분리

- 셀마다 UISwitch를 매번 생성하지 않고 프로퍼티로 관리
- configure 메서드에서 상태만 갱신하도록 변경

kim-eden commented on PodoIt/Scenes/Setting/View/SettingViewController.swift:
```diff
@@ -0,0 +1,83 @@
+  // MARK: - updateTableHeaderFooterHeight
+
+  // header,footer 의 레이아웃을 기반으로 높이 계산 (UIView의 오토 레이아웃으로 적용)
+  // 중복 코드가 많아서 하나로 사용
+  private func updateTableHeaderFooterHeight(_ view: UIView, assign: (UIView) -> Void) {
+    let size = view.systemLayoutSizeFitting(
+      CGSize(width: tableView.bounds.width, height: UIView.layoutFittingCompressedSize.height)
+    )
+    view.frame.size.height = size.height
+    assign(view)
+  }
+}
```
중복 코드 줄여서 관리하는거 좋네요!

kim-eden commented on PodoIt/Scenes/Setting/View/SettingViewController.swift:
```diff
@@ -0,0 +1,83 @@
+    super.viewDidLayoutSubviews()
+    updateTableHeaderFooterHeight(headerView, assign: { tableView.tableHeaderView = $0 })
+    updateTableHeaderFooterHeight(footerView, assign: { tableView.tableFooterView = $0 })
+  }
```
viewDidLayoutSubviews() 할때마다 헤더/푸터가 매번 재할당 되는 것 같은데 초기 1회만 설정해줘도 좋을 것 같아요!

Gwangyong replied:
리뷰 감사합니다. didSetHeaderFooter 플래그를 통해서 1회만 호출되도록 수정했습니다👍

kim-eden reviewed (approved):
고생하셨습니다~!

Gwangyong committed:
fix: #4 - header/footer 레이아웃 계산 1회만 실행되도록 수정

- viewDidLayoutSubviews에서 반복 호출되던 문제 해결
- didSetHeaderFooter 플래그를 통해서 최초 1회만 높이 계산

Gwangyong committed:
Merge remote-tracking branch 'origin/develop' into feat/4-settings-ui
